### PR TITLE
Add truncation for activity exception event

### PR DESCRIPTION
**Pull request by vishweshbankwar**

### CODE REVIEW

**Review by TimothyMothra — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs [hunk 1]
```diff
@@ -232,12 +232,12 @@ internal static MonitorBase GetExceptionDataDetailsOnTelemetryItem(IEnumerable<K
                 return null;
             }
 
-            TelemetryExceptionDetails exceptionDetails = new(exceptionMessage)
+            TelemetryExceptionDetails exceptionDetails = new(exceptionMessage.Truncate(SchemaConstants.ExceptionDetails_Message_MaxLength))
             {
-                Stack = exceptionStackTrace,
+                Stack = exceptionStackTrace.Truncate(SchemaConstants.ExceptionDetails_Stack_MaxLength),
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs:237]: Did we test this manually and does the ingestion accept the truncated stack?

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs:237]: it does - the only thing that I am noticing is, it is not able to parse the stack like below
image

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs:237]: This is different from the experience from exception captured from Logs. Should we truncate to the stack to last carriage return on the trimmed length?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs:237]: @rajkumar-rangaraj would this algorithm help here?
https://github.com/Azure/azure-sdk-for-net/blob/708c95c25b43192eed35fa657b662595a0dbd765/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryExceptionDetails.cs#L59-L61

I had forgotten about this until your last comment.

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs:237]: cannot use that as we don't have access to exception object in this case.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs (hunk 1 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs:237]: Keeping the truncation as is for now. Will bring this up in UI testing.

**Pull request merged**